(#2) Adds Initial Examples (#3)
chocolatey/ccm-deployment-examples: deployments/upgrading-license-daily/README.md

# Upgrading Chocolatey License Regularly

## Purpose

> This package is a basic example of deploying a package, in this case `chocolatey.license`, on a schedule.

Reviewer: ```suggestion
These Deployment Plans are basic examples of deploying a package, in this case `chocolatey.license`, on a schedule.
```

> These Deployment Plans are basic examples of deploying a package, in this case `chocolatey.license`, on a schedule.

Reviewer 2: ```suggestion
These Deployment Plans are basic examples of deploying a package on a schedule. In this case `chocolatey.license`.
```

https://github.com/chocolatey/ccm-deployment-examples/blob/b79340df0363be0fcda3192828cef96095320d3d/deployments/upgrading-license-daily/Deployment.json#L1-L22

## Usage

> The only deployment step contained upgrades (installing if not present) the `chocolatey.license` package (which is we recommend creating when using C4B environments, to ensure you can push license updates out). If you don't currently have a license package, you can create one by following [the instructions available at docs.chocolatey.org](https://docs.chocolatey.org/en-us/c4b-environments/azure/license-update#creating-a-new-license-package).
> 
> There is a second example deployment that would run weekly.

Reviewer 2: ```suggestion
We recommend creating a `chocolatey.license` package when using Chocolatey for Business environments. This ensures you can push out license changes to your client computers.

The Deployment Step upgrades, or installs if not present, the `chocolatey.license` package . If you don't currently have a license package, you can create one by following [the instructions](https://docs.chocolatey.org/en-us/c4b-environments/azure/license-update#creating-a-new-license-package).

There is a second example Deployment Plan that would run weekly.
```

Author: ```suggestion
We recommend creating a `chocolatey.license` package when using Chocolatey for Business environments. This ensures you can push out license changes to your client computers.

The Deployment Step upgrades, or installs if not present, the `chocolatey.license` package. If you don't currently have a license package, you can create one by following [the instructions](https://docs.chocolatey.org/en-us/c4b-environments/azure/license-update#creating-a-new-license-package).

There is a second example Deployment Plan that would run weekly.
```

https://github.com/chocolatey/ccm-deployment-examples/blob/b79340df0363be0fcda3192828cef96095320d3d/deployments/upgrading-license-daily/Deployment-Weekly.json#L3-L5

> There are [more options available](https://docs.chocolatey.org/en-us/central-management/usage/website/deployments#recurring-deployments) for scheduling. You can see the required values for your current version of Chocolatey Central Management in the provided swagger document.

Reviewer 2: Where is the swagger document?

Author: It is hard to provide an answer to this, as we only make it available on individual instances so the FQDN will change per-user. That said, I could link [here](https://docs.chocolatey.org/en-us/central-management/usage/api/#description) or add instructions here.

> After importing the deployment, you will need to set both the target groups (we recommend using the `All Computers` group) and the `Start Date Time`. You may also want to change the deployment name to something more meaningful.

Reviewer 2: ```suggestion
After importing the Deployment Plans, you will need to set the Target Groups and `Start Date Time` for each. We recommend using the `All Computers` Target Group. You may also want to change the Deployment Plan name to something more meaningful.
```

> ### Importing the Deployment
> 
> - Download the `Deployment.json` from this directory (or the `Deployment-Weekly.json` example)
> 
> - Login to your CCM instance
> 
> - Navigate to the `Deployments` tab (on the left navigation bar) and click `Import Deployment`  
>     ![The Deployments selection on the left nav bar, and the Import Deployment button](/images/DeploymentTab.jpg)
> 
> - Click `Browse` and select the downloaded json file  
>     ![The Import Deployment Modal](/images/DeploymentImportModal.jpg)
> 
> - Click `Import`
> 
> For further details, please see the [documentation at docs.chocolatey.org](https://docs.chocolatey.org/en-us/central-management/usage/examples/deployments).

Reviewer 2: ```suggestion
### Importing the Deployment Plans

- Download the `Deployment.json` from this directory (or the `Deployment-Weekly.json` example).

- Login to your Chocolatey Central Management instance.

- Navigate to the `Deployments` tab (on the left navigation bar) and click `Import Deployment`.
    ![The Deployments selection on the left nav bar, and the Import Deployment button](/images/DeploymentTab.jpg)

- Click `Browse` and select the downloaded `.json` file.
    ![The Import Deployment Modal](/images/DeploymentImportModal.jpg)

- Click `Import`.

For further details, please see the [documentation](https://docs.chocolatey.org/en-us/central-management/usage/examples/deployments).
```

### Setting the Start Date Time

> Before you can run the deployment, you must set the `Start Date Time` to a time at least 30 minutes in the future.

Reviewer 2: ```suggestion
Before you can run the Deployment Plan, you must set the `Start Date Time` to a time at least 30 minutes in the future.
```

Before you can run the Deployment Plan, you must set the `Start Date Time` to a time at least 30 minutes in the future.

> To do so, click on the clock icon next to the `Start Date Time` field. When you have selected an appropriate time, hit `Save`.

Reviewer 2: ```suggestion
To do so, click on the clock icon next to the `Start Date Time` field. When you have selected an appropriate time, click `Save`.
```

### Setting the Target Group

> You will also need to set a target group for the deployment. 
> 
> To do so:

Reviewer 2: ```suggestion
You will also need to set a Target Group for the Deployment Step:
```

> - Mouse-over the deployment step and click the `Edit` button:
>     ![Editing a Deployment Step](/images/EditDeploymentStep.jpg)
> 
> - Navigate to the `Select Target Groups` panel
> 
> - Highlight your preferred groups and click the `>` button to move them into `Selected Groups`
>     ![Selecting Target Groups](/images/SelectingTargetGroups.jpg)
> 
> - Click `Save`

Reviewer 2: ```suggestion
- Mouse-over the Deployment Step and click the `Edit` button.
    ![Editing a Deployment Step](/images/EditDeploymentStep.jpg)

- Navigate to the `Select Target Groups` panel.

- Highlight your preferred groups and click the `>` button to move them into `Selected Groups`.
    ![Selecting Target Groups](/images/SelectingTargetGroups.jpg)

- Click `Save`.
```
